pyramid_rpc/jsonrpc.py

```diff
@@ -256,6 +270,48 @@ def __call__(self, context, request):
         return getattr(request, 'rpc_method') == self.method
 
 
+class BatchedRequestPredicate(object):
+    def __init__(self, val, config):
+        self.val = val
+
+    def text(self):
+        return 'jsonrpc batched request = %s' % self.val
+
+    phash = text
+
+    def __call__(self, context, request):
+        if self.val:
+            return hasattr(request, 'batched_rpc_requests')
+
+
+def batched_request_view(request):
+    json_response = []
+    response = request.response
+    for rpc_request in request.batched_rpc_requests:
+        body = json.dumps(rpc_request).encode(request.charset)
+        subrequest = Request.blank(path=request.path,
+                                   environ=request.environ,
+                                   base_url=request.application_url,
+                                   headers=request.headers,
+                                   POST=body,
+                                   charset=request.charset)
+        subresponse = request.invoke_subrequest(subrequest, use_tweens=True)
+        if subresponse.json_body != '':
+            json_response.append(subresponse.json_body)
+    if json_response:
+        # use charset and content-type from last subresponse
+        response.charset = subresponse.charset
+        response.content_type = subresponse.content_type
+        # will automatically be encoded
+        response.json_body = json_response
+    else:
+        # if we would send an empty list, instead send nothing
+        # per JSON-RPC: http://www.jsonrpc.org/specification#batch
+        response.content_type = 'text/plain'
+        response.body = b''
```

Comment:
What would the options be for it?
If it were `{}` then we'd want to change the content type as well, right?